tests/conftest.py

```
@@ -0,0 +1,253 @@
+    Handles the safe, automatic creation and teardown of a dedicated test database
+    for the entire test session by deriving the name from the main DATABASE_URL.
+    """
+    # 1. Load the main .env file to get the development DB URL
+    # This ensures we have a single source of truth for the DB configuration.
+    load_dotenv()
+    main_db_url = os.getenv("POSTGRES_SERVER")
+
+    if not main_db_url:
+        raise ValueError(
+            "FATAL: DATABASE_URL not found in your .env file. Cannot derive test database URL."
+        )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)